### PR TITLE
refactor models list, delete products table, remake tests

### PR DESCRIPTION
**Pull request by Filiczini**
## Description

Refactored models-list.js component with graphql, delete products-table page (unused on project).

### Checklist

- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue - https://github.com/ita-social-projects/horondi_client_fe/issues/994

### CODE REVIEW

File: src/pages/home/models-list/models-list.js [hunk 1]
```diff
@@ -1,41 +1,35 @@
...
   const onShowModels = () => {
     setIsModelsVisible(!isModelsVisible);
   };
 
-  if (modelsLoading) {
+  const { loading, error } = useQuery(getAllModelsQuery, {
+    onCompleted: (data) => setModels(data.getAllModels.items)
+  });
+
+  if (loading || error) {
```
Comment (maarekua) [inline src/pages/home/models-list/models-list.js:26]: Its kinda strange showing Loader when we have an error.

File: src/pages/home/models-list/models-list.js [hunk 2]
```diff
@@ -1,41 +1,35 @@
...
     return <Loader />;
   }
-
   return (
     <div className={styles.root} data-section-style='light' id='models'>
       <div className={styles.modelsWrapper}>
-        {models.map((model) => (
+        {models?.map((model) => (
```
Comment (maarekua) [inline src/pages/home/models-list/models-list.js:32]: I think we don't need ?. operator cause if something went wrong we will not reach this code.

File: src/pages/home/models-list/operations/getModelsByCategory.queries.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { gql } from '@apollo/client';
+
+export const getModelsByCategoryQuery = gql`
```
Comment (maarekua) [inline src/pages/home/models-list/operations/getModelsByCategory.queries.js:3]: Place both queries in one file.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/pages/home/models-list/models-list.js (hunk 2 above)
Comment (Filiczini) [reply src/pages/home/models-list/models-list.js]: done

File: src/pages/home/models-list/operations/getModelsByCategory.queries.js (hunk 3 above)
Comment (Filiczini) [reply src/pages/home/models-list/operations/getModelsByCategory.queries.js]: Second querie deleted. Unusable querie.

**Review by olga-gchak — DISMISSED** [review]

**Review by ElitKondor — APPROVED** [review]

**Pull request merged**